lib/ansible/executor/task_result.py

```diff
@@ -69,3 +72,33 @@ def _check_key(self, key):
                 if isinstance(res, dict):
                     flag |= res.get(key, False)
             return flag
+
+    def clean_copy(self):
```

Comment:
[note for potential future changes that doesnt need to be addressed now for this PR]

We have a couple of places where we do some variant of 'copy a dict, but then remove certain keys'. Might be worthwhile to generalize and add to utils and use.

Reply:
this should end up doing more than that, also creating mock task/play to avoid direct refs .. but that should not preclude such function